modules/nvidia_plugin/src/cuda_plugin.cpp

```diff
@@ -228,8 +228,10 @@ InferenceEngine::Parameter Plugin::GetMetric(const std::string& name,
     } else if (METRIC_KEY(IMPORT_EXPORT_SUPPORT) == name) {
         IE_SET_METRIC_RETURN(IMPORT_EXPORT_SUPPORT, true);
     } else if (METRIC_KEY(DEVICE_ARCHITECTURE) == name) {
-        // TODO: return device architecture for device specified by DEVICE_ID config
-        std::string arch = "CUDA";
+        const std::string deviceId = _cfg.Get(CONFIG_KEY(DEVICE_ID));
+        CUDA::Device device{std::stoi(deviceId)};
+        const auto& props = device.props();
+        std::string arch = "NVIDIA: " + std::string(props.name);
```

Comment:
looks like we define architecture as a product name, is it ok? I suppose product with different names can have the same architecture.
This property is used in caching to differentiate compiled blob for incompatible architectures.

Reply:
> looks like we define architecture as a product name, is it ok? I suppose product with different names can have the same architecture. This property is used in caching to differentiate compiled blob for incompatible architectures.

1. Currently we have the same architecture (`CUDA`) for different hw, which can lead to caching problem due to different hw capabilities on the device (supported precisions, for example). 

2. PR aligns behavior between Intel GPU plugin and NVIDIA plugin. Is this comment also applicable for Intel GPU plugin? @vladimir-paramuzov 

3. Do we have a requirements to support caching between different product with the same architecture (expeccially for contrib)?

Reply:
Replaced by device capabilities. But caching would work on different platforms in PROXY/HETERO scenario due to Intel GPU architecture mismatch

Reply:
> Intel GPU architecture mismatch

Is it because for NVIDIA GPU GXF major is 0?
```
    } else if (name == ov::device::architecture) {
        std::stringstream s;
        s << "GPU: ";
        if (device_info.gfx_ver.major == 0 && device_info.gfx_ver.minor == 0) {
            s << device_info.dev_name;
        } else {
            s << "v" << static_cast<int>(device_info.gfx_ver.major)
              << "." << static_cast<int>(device_info.gfx_ver.minor)
              << "." << static_cast<int>(device_info.gfx_ver.revision);
        }
        return decltype(ov::device::architecture)::value_type {s.str()};
    } else {
```
Maybe we can extract NVIDIA device capabilities in Intel GPU plugin OpenCL?

Reply:
@ilya-lavrenov it can be done using https://registry.khronos.org/OpenCL/extensions/nv/cl_nv_device_attribute_query.txt extension